lib/ldclient-rb/impl/integrations/file_data_source.rb

```diff
@@ -40,6 +40,9 @@ def initialize(data_store, data_source_update_sink, logger, options={})
           @poll_interval = options[:poll_interval] || 1
           @initialized = Concurrent::AtomicBoolean.new(false)
           @ready = Concurrent::Event.new
+
+          @version_lock = Mutex.new
```

Comment:
This does result in some behavior change.

If you modify a single flag, all of the flags are going to appear to have changed since we are incrementing all of their versions. This is what the Java SDK does.

Reply:
It is probably fine, but I know we have a few different ways this behaves.

Node: Only update the version if it thinks it changed, but the logic may not be the best.
Dotnet: One shared version number that applies to all flags.

And some that had this previous behavior. Something we may want to write a spec for at some point.

Both are technically things that *could* happen with a real client.